internal/contour/listener.go

```diff
@@ -194,6 +194,19 @@ func (lc *ListenerCache) httpsAddress() string {
 	return DEFAULT_HTTPS_LISTENER_ADDRESS
 }
 
+func (lc *listenerCache) HostHasTls(host string) bool {
+	for _, listener := range lc.Values() {
```

Comment:
Please don't use lc.Values for this. Any information that you get out of lc.Values will be present in the other caches maintained by the listenerCache.